tools/build/Library/InstallUnpacker/classes/ConfigurationValidator.php

```diff
@@ -0,0 +1,155 @@
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+class ConfigurationValidator
+{
+    /**
+     * Check whether download, unzip & replace of latest PS version can be carried out.
+     *
+     * @return string[] array of issues. Empty if download can be performed.
+     */
+    public function testSystemCanPerformDownloadUnzipAndReplace()
+    {
+        $errors = [];
+
+        if (!$this->testCurl() && !$this->testFopen()) {
+            $errors[] = 'You need allow_url_fopen or cURL enabled for automatic download to work.';
+        }
+
+        $systemErrors = $this->testFilesystemCapabilities();
+        if (false === empty($systemErrors)) {
+            $errors[] = sprintf('Cannot manipulate file on this system: %s', implode('; ', $systemErrors));
+        }
+
+        if (false === class_exists('ZipArchive')) {
+            $errors[] = 'Cannot unzip files as php class ZipArchive is not available.';
+        }
+
+        // @todo: be able to use fallback util directories
+        $downloadDirPath = __DIR__.DIRECTORY_SEPARATOR.'download';
+        if (is_dir($downloadDirPath) || file_exists($downloadDirPath)) {
+            $errors[] = "Directory 'download' already exists.";
+        }
+        $downloadDirPath = __DIR__.DIRECTORY_SEPARATOR.'extracted';
+        if (is_dir($downloadDirPath) || file_exists($downloadDirPath)) {
+            $errors[] = "Directory 'extracted' already exists.";
+        }
+
+        return $errors;
+    }
+
+    /**
+     * @return bool
+     */
+    public function testFopen()
+    {
+        return in_array(ini_get('allow_url_fopen'), array('On', 'on', '1'));
+    }
+
+    /**
+     * @return bool
+     */
+    public function testCurl()
+    {
+        return extension_loaded('curl');
+    }
+
+    /**
+     * Test whether files and directories can be manipulated by php on given system
+     * by performing multiple file operations
+     *
+     * @return string[]
+     */
+    public function testFilesystemCapabilities()
+    {
+        // choose a random available directory name
+        $dirPath = $this->getRandomDirectoryPath();
+
+        $randomNameIsActuallyAlreadyUsed = (is_dir($dirPath) || file_exists($dirPath));
+        if ($randomNameIsActuallyAlreadyUsed) {
+            // retry another random directory name
+            $dirPath = $this->getRandomDirectoryPath();
+
+            $randomNameIsActuallyAlreadyUsed = (is_dir($dirPath) || file_exists($dirPath));
+            if ($randomNameIsActuallyAlreadyUsed) {
+                throw new \RuntimeException('Failed to find available directory name');
+            }
+        }
+
+        // create directory
+        $createDirectoryResult = @mkdir($dirPath);
+        if (false === $createDirectoryResult) {
+            return ['Cannot create directories'];
+        }
+
+        // create file in it
```

Comment:
instead of comments, why not wrap every portion of code in a well-named function?

Reply:
Most of these tests are 1 line of code. Is that worth it to write 6 private functions of 1 line each to be used once each ?